astropy/time/core.py

```diff
@@ -293,6 +293,10 @@ def _get_time_fmt(self, val, val2, format, scale):
                        if issubclass(cls, TimeUnique)]
             err_msg = ('any of the formats where the format keyword is '
                        'optional {0}'.format([name for name, cls in formats]))
+            # AstropyTime is a pseudo-format that isn't in the TIME_FORMATS registry,
+            # but try to guess it at the end.
```

Comment:
By inserting before `0`, this will be the *first* format tried. Was that intended? If it should be the last one, it would be
`formats.append(('astropy_time', TimeAstropyTime))`.